### ISSUE TITLE
Calling `secureHashFile(filename: string)` loads the whole file in memory

### ISSUE THREAD

**Issue by irdassis** [nim-lang/Nim#15997]
The function `secureHashFile(filename: string)` reads the whole file into memory and then computes the SHA1 hash, this causes OOM if the file is big enough.

### Example
Get a big enough file:
```sh
cat /dev/random > bigfile.txt
``` 

Try to compute the sha1 hash using the standard library function
```nim
import std/sha1

echo $secureHashFile("bigfile.txt")
```

### Current Output
If you don't have enough memory you will get a crash with OOM.

### Expected Output

The sha1 hash of the file.

### Possible Solution

I wrote my own version of  `secureHashFile(filename: string)` that uses a fixed 8KB buffer. It is probably not the most idiomatic Nim but it works.

```nim
import std/sha1

proc secureHashFileBuffered*(filename: string): SecureHash =
  var state = newSha1State()

  let f = open(filename)

  const bufferLength = 8192
  var buffer = newString(bufferLength)
  while true:
    let length = readChars(f, buffer, 0, bufferLength)
    if length == 0:
        break
    buffer.setLen(length)
    state.update(buffer)
    if length != bufferLength:
        buffer.setLen(bufferLength)
  close(f)

  SecureHash(state.finalize())

if isMainModule:
    assert secureHashFileBuffered("testfile.txt") == secureHashFile("testfile.txt")
    echo $secureHashFileBuffered("testfile.txt"), " == ", secureHashFile("testfile.txt")
```

### Additional Information
I tried with version `1.2.6` but the code of the library is the same in the devel branch so I don't expect the result to be different.

**Comment by timotheecour** [comment]
@irdassis please make a PR based on your code snippet

**Comment by ringabout** [comment]
closed by #16006
